### ISSUE TITLE
Notify the user when trying to read a non-existing column

### ISSUE THREAD

**Issue by vasil-pashov** [man-group/ArcticDB#621]
**Is your feature request related to a problem? Please describe.**
When trying to read a column via `read(symbol, columns=[column_name])` and the column is not in the symbol Arctic would return an empty dataframe. This happens both for DataFrames and for Series. Example:
```python
from arcticdb import Arctic
import pandas as pd
import numpy as np

ac = Arctic('lmdb://temp/test')

# pd.Series
if 'series' not in ac.list_libraries():
        ac.create_library('series')
lib = ac['series']
lib.write("s", pd.Series([1,2,3], index=pd.date_range("2018", periods=3)))

result_no_columns = lib["s"].data
assert isinstance(result_no_columns, pd.Series)
assert len(result_no_columns) == 3

result_with_columns = lib.read("s", columns=["x"]).data
assert isinstance(result_with_columns, pd.DataFrame)
assert len(result_with_columns) == 0

# pd.DataFrame
df = pd.DataFrame(np.ones((4, 3))*1, index=pd.date_range('1/1/2023', periods=4, freq="D"), columns=list('ABC'))
lib.write("df",  df)

df_result_with_columns = lib.read("df", columns=["x"]).data
assert isinstance(df_result_with_columns, pd.DataFrame)
assert len(result_with_columns) == 0
```
Note also that that `result_with_columns = lib.read("s", columns=["x"]).data` returns a DataFrame.

**Describe the solution you'd like**
Ideally throw an exception when trying to read nonexistent columns or at least notify the user somehow that there are no such columns.

**Describe alternatives you've considered**

**Additional context**

**Comment by dragman** [comment]
It's more about the elimination of surprising behaviour.

a) returning a different type, did I ask for a series or a dataframe?
b) the behaviour is different between a series with a datetimeindex and one with an integer index, integer index returns an unchanged series, datetime index returns an empty dataframe

`
s = pd.Series([1,2,3])
lib.write("test", s)
result_with_column = lib.read("test", columns=["x"])
pd.testing.assert_series_equal(result_with_column.data, s)
`

**Comment by jamesmunro** [comment]
Definitely agree with a) and b), but also not sure about raising an exception.  I'd like to see a wider set of use-cases.

**Comment by dragman** [comment]
So the exception raising idea came from consistency with pandas behaviour.  

i.e.  `a_series.reindex(columns=[])` would raise.

I appreciate it's not quite the same use case here and agree it's not obvious that raising is best course of action, but I do think asking for columns from a series is an indication of a logic error that might help the programmer if handled more predictably.

**Comment by jamesmunro** [comment]
Agreed.  We do also need to support methods like `read_batch` and future join functionality without to many changes in behavior, hence my doubt in general.
